doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,605 @@
+and route the packet over an intra-AS path matching the provided FABRID policy.
+All intra-AS paths are configured by the AS operator, and are provided to the border routers by the local control service.
+
+Processing at the endhost
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To be able to send a FABRID packet, the endhost has to choose a path that supports its path and policy constraints.
+A detailed explanation on how endhost applications can find such paths is given in the section :ref:`Exposing policies to the end hosts <endhost_policy_selection>`.
+Once a path has been found, with specific policies for each hop in the path, the path and an array containing one policy per hop is given to the FABRID snet implementation.
+The snet implementation then constructs the FABRID packet by automatically requesting the necessary DRKeys and computing the hop validation fields.
+The packet can then be sent to the border router for further forwarding.
+A receiving endhost can recompute the path validator to verify that the packet was forwarded over this path.
+
+Modifications of the control plane
+------------------------------------
+
+Control service
+^^^^^^^^^^^^^^^^^
+
+The control service for FABRID is responsible for maintaining the by the AS-operator configured FABRID policies, intra-AS paths,
+and making them accessible for the routers, the endhosts and other remote control services.
+The policies are defined between interface pairs and for the last AS on the path also per interface - IP range pair.
+Through gRPC, border routers can query the control service for the list of supported policies,
+as well as the mapping from policies to MPLS labels.
+Policies are disseminated to remote ASes through PCBs, which clients in the AS can query from their Path Servers.
+This policy information can also be requested directly from remote ASes over gRPC.
+
+The control service introduces a FABRID service with the following endpoints where *intra-AS* means it can be reached
+from the local AS and *inter-AS* means it can be reached from a remote AS:
+
+- GetMPLSMapIfNecessary (intra-AS)
+    Is used by the router to retrieve the MPLS map for the intra-AS paths.
+    The map is only returned if the router does not have an up to date MPLS map.
+- GetRemotePolicyDescription (intra-AS)
+    Is used by the endhosts of the local AS to request the policy description of a policy identifier for a remote AS.
+- GetSupportedIndicesMap (inter-AS, intra-AS)
+    Returns the per interface-pair supported FABRID indices.
+- GetIndexIdentifierMap (inter-AS, intra-AS)
+    Returns a map that maps identifiers to indicies which can then be used for sending FABRID packets.
+- GetLocalPolicyDescription (inter-AS, intra-AS)
+    Is used to request the policy description of a policy identifier for the local AS.
+
+Important data structures
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following list explains the most important data structures used in the FABRID service:
+
+- SupportedIndicesMap
```

Comment:
As the maps in this subsection are anyway not formally introduced, maybe just write "Supported indices map"?